pip/req/req_install.py

```diff
@@ -312,7 +312,13 @@ def build_location(self, build_dir):
             # package is not available yet so we create a temp directory
             # Once run_egg_info will have run, we'll be able
             # to fix it via _correct_build_location
-            self._temp_build_dir = tempfile.mkdtemp('-build', 'pip-')
+            #
+            # NOTE: Some systems have /tmp as a symlink which confuses
```

Comment:
Do you want me to:

1. remove the comment completely?
2. remove the marker "NOTE: " only?
3. merge the comment with the one above on line 314?

Let me know, I can do whatever is preferred here.
It's important to draw attention to the reason for realpath, though, to avoid future regressions. This bug resurfaced multiple times through pip history and we want to avoid that. :-)